tests/integration/graph/test_graph_varia.py

```diff
@@ -658,3 +658,115 @@ def load_chart(n_clicks):
     scripts = dash_dcc.driver.find_elements(By.CSS_SELECTOR, "script")
     assert findSyncPlotlyJs(scripts) is None
     assert findAsyncPlotlyJs(scripts) is None
+
+
+def test_shapes_not_lost(dash_dcc):
+    # See issue #879 and pr #905
+    app = dash.Dash(__name__)
+
+    fig = {"data": [], "layout": {"dragmode": "drawrect"}}
+    graph = dcc.Graph(id="graph", figure=fig)
+
+    app.layout = html.Div(
+        [
+            graph,
+            html.Br(),
+            html.Button(id="button", children="Clone figure"),
+            html.Div(id="output", children=""),
+        ]
+    )
+
+    app.clientside_callback(
+        """function clone_figure(_, figure) {
+            let new_figure = {...figure};
+            let shapes = new_figure.layout.shapes || [];
+            return [new_figure, shapes.length];
+        }
+        """,
+        [Output("graph", "figure"), Output("output", "children")],
+        [Input("button", "n_clicks")],
+        [State("graph", "figure")],
+    )
+
+    dash_dcc.start_server(app)
+    button = dash_dcc.wait_for_element("#button")
+    time.sleep(1)
```

Comment:
Is there a way we can avoid `sleep` here? Elsewhere in graph tests we use things like
`dash_dcc.wait_for_element("#my-graph:not([data-dash-is-loading])")`
(ie the renderer thinks the graph is done) or
`dash_dcc.wait_for_element("#my-graph .main-svg")`
(ie plotly.js has created some of the structure we know the graph should have)

Reply:
Managed to get rid of these!